### ISSUE TITLE
Depends On does not work when the parent task was deleted and then re-created

### ISSUE THREAD

**Issue by gtrubach-iballistix** [Azure/Batch#129]
#### Problem Description

Hello,

I'm not sure if it's a bug or it's by design, but the situation is the following:
* Suppose we have two tasks A and B, and B depends on A.
* We created both tasks and now A is running while B is waiting for A.
* Now A is completed, B is running. B competes.
* Then we want to re-create tasks A and B with the same IDs. We deleted both A and B and re-created them in the same way as before - A is a parent, B depends on A.
* Now Batch runs both A and B simultaneously even though A is still running.

Is it possible to make B to wait for A after task re-creation?

#### Steps to Reproduce
0. Ensure, that your pool can run multiple tasks in parallel.
1. Create two tasks A and B and make B depend on A.
2. Wait for both tasks completion. Make a notice, that B won't start till A completes.
3. Delete both tasks A and B.
4. Create tasks A and B again with the same IDs and dependence as in the first step.
5. Wait till both tasks start.

#### Expected Results
Task B waits till tasks A completes.

#### Actual Results
Both tasks A and B started and running simultaneously.

**Comment by martinjoshua** [comment]
Yes! I have experienced this same issue and opened a case with support (#2201210040007594). 

Unfortunately, the case ended with the Batch team being unable to see in the logs where this occurred... but it did happen. I had to delete the set of already-completed tasks, and then manually add them back one by one (waiting until the dependency completed before adding the next.) Without knowing the inner workings of Batch, it sure seems like there is a task ID and exit code being cached somewhere, so that even after deleting and re-adding, the Batch scheduler sees the (cached) task as already completed and continues scheduling...
